### PR TITLE
Fix problems with concurrent updates in Compass

### PR DESCRIPTION
**Pull request by Maladie**
**Description**

Changes proposed in this pull request:

- Add retry on specific error code to compass client.

**Related issue(s)**

https://github.com/kyma-project/kyma/issues/12583

### CODE REVIEW

File: tests/fast-integration/compass/client.js [hunk 1]
```diff
@@ -251,4 +251,19 @@ class DirectorClient {
...
     DirectorClient
-};
\ No newline at end of file
+};
+
+const retry = async (fn) => {
+        try {
+            return await fn()
+        } catch (err) {
+            if (err.errors && err.errors[0].extensions.errorCode === 35) {
```
Comment (franpog859) [inline tests/fast-integration/compass/client.js:260]: I think we could add a comment above this line with the link to the commit in Compass (`https://github.com/kyma-incubator/compass/commit/86aa036ee48cf207230e606ee18b3cb8edbbc1e4`) so some future developers will know what the helm is this `35` :D

File: tests/fast-integration/compass/client.js [hunk 2]
```diff
@@ -251,4 +251,19 @@ class DirectorClient {
 module.exports = {
     DirectorConfig,
     DirectorClient
-};
\ No newline at end of file
+};
+
+const retry = async (fn) => {
```
Comment (franpog859) [inline tests/fast-integration/compass/client.js:256]: Maybe we could name this function `retryOnConcurrentUpdateError` or something like this to add some transparency. What do you think?

File: tests/fast-integration/compass/client.js [hunk 3]
```diff
@@ -78,18 +78,19 @@ class DirectorClient {
...
-            if(resp.data.errors) {
+            const resp = await retryOnConcurrentUpdate(() => axios.post(url, body, params), 10);
+            if (resp.data.errors) {
                 debug(resp);
                 throw resp.data;
             }
             return resp.data.data.result;
-        } catch(err) {
+        } catch (err) {
+            console.dir(err)
```
Comment (franpog859) [inline tests/fast-integration/compass/client.js:88]: Remove this log

**Review by franpog859 — APPROVED** [review]
Looks good 💪

**Review by pbochynski — APPROVED** [review]

**Pull request merged**